src/cleanvision/dataset/utils.py

```diff
@@ -0,0 +1,33 @@
+from __future__ import annotations
+
+from typing import List, Optional, TYPE_CHECKING
+
+from cleanvision.dataset import Dataset
+from cleanvision.dataset.folder_dataset import FolderDataset
+from cleanvision.dataset.hf_dataset import HFDataset
+from cleanvision.dataset.torch_dataset import TorchDataset
+
+if TYPE_CHECKING:
+    import datasets
+    from torchvision.datasets.vision import VisionDataset
+
+
+def build_dataset(
+    data_path: Optional[str] = None,
+    filepaths: Optional[List[str]] = None,
+    hf_dataset: Optional["datasets.Dataset"] = None,
+    image_key: Optional[str] = None,
+    torchvision_dataset: Optional["VisionDataset"] = None,
+) -> Dataset:
+    if data_path:
+        return FolderDataset(data_folder=data_path)
+    elif filepaths:
+        return FolderDataset(filepaths=filepaths)
```

Comment:
definitely needs to be covered by a unit test

Reply:
Added